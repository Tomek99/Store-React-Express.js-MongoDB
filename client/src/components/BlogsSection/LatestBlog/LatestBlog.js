import React from "react";
import styles from "./LatestBlog.module.scss";
import PropTypes from "prop-types";

function LatestBlog({ item, text }) {
  const { id, imageUrl, title, addedBy } = item;

  return (
    <div key={id} className={styles.singleBlog}>
      <div className={styles.zoomImage}>
        <img src={imageUrl} alt={title} />
      </div>
      <div className={styles.textSection}>
        <h2>{title}</h2>
        <h4>{addedBy}</h4>
        <p>{text}</p>
      </div>
    </div>
  );
}

LatestBlog.propTypes = {
  id: PropTypes.string,
  url: PropTypes.string,
  title: PropTypes.string,
  addedBy: PropTypes.string,
  text: PropTypes.string,
};
export default LatestBlog;
