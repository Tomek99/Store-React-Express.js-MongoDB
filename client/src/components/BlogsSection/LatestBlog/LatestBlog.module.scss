.singleBlog {
  display: flex;
  flex-direction: column;
  gap: 10px;
  width: 360px;
  height: auto;
  overflow: hidden;
  transition: 0.3s linear;

  .zoomImage {
    overflow: hidden;
    cursor: pointer;

    img {
      transition: 0.3s linear;
      width: 100%;
    }
  }

  .textSection {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    gap: 10px;
    margin: 10px;
    color: var(--text-color);
    h2 {
      transition: 0.3s linear;
      font-size: 2rem;
      cursor: pointer;
    }

    h4 {
      font-size: 1.6rem;
      font-weight: 400;
    }

    p {
      font-size: 1.3rem;
    }
  }
}

.singleBlog:hover img {
  transform: scale(1.1);
}
