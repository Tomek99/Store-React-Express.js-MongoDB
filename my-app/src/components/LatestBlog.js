import React from 'react'

function LatestBlog(props) {
    return (
        <div key={props.key} className="singleBlog">
            <img src={props.url} alt={props.title} />
            <h2>{props.title}</h2>
            <h4>{props.addedBy}</h4>
            <p>{props.text}</p>
            <button>Read more</button>
        </div>
    )
}

export default LatestBlog