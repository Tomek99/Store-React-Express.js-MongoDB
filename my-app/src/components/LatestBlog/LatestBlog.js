import React from 'react'


function LatestBlog(props) {
    return (
        <div key={props.key} className="singleBlog">
            <div className='zoomImage'>
                <img src={props.url} alt={props.title} />
            </div>
            <div className='textSection'>
                <h2>{props.title}</h2>
                <h4>{props.addedBy}</h4>
                <p>{props.text}</p>
                <button>Read more</button>
            </div>
        </div>
    )
}

export default LatestBlog