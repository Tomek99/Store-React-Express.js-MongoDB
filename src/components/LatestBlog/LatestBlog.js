import React from "react";
import Button from "../Buttons/ButtonBlogs";
import styles from "./LatestBlog.module.scss";

function LatestBlog(props) {
  return (
    <div key={props.id} className={styles.singleBlog}>
      <div className={styles.zoomImage}>
        <img src={props.url} alt={props.title} />
      </div>
      <div className={styles.textSection}>
        <h2>{props.title}</h2>
        <h4>{props.addedBy}</h4>
        <p>{props.text}</p>
        <Button text="read more" />
      </div>
    </div>
  );
}

export default LatestBlog;
