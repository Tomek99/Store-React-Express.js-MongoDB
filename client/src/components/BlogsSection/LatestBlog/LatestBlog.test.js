import { render, screen } from "@testing-library/react";
import LatestBlog from "./LatestBlog";

test("should render LatestBlog component", () => {
  const item = {
    title: "Lorem ipsum",
    text: " Lorem ipsum lorem ipsum",
    addedBy: "Tomek S",
    imageUrl: "images/blog-1.jpeg",
  };

  const { container } = render(<LatestBlog item={item} />);

  expect(container).toMatchSnapshot();
});
