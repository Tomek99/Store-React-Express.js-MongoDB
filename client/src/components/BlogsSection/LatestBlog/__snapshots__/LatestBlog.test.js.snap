// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LatestBlog component 1`] = `
<div>
  <div
    class="singleBlog"
  >
    <div
      class="zoomImage"
    >
      <img
        alt="Lorem ipsum"
        src="images/blog-1.jpeg"
      />
    </div>
    <div
      class="textSection"
    >
      <h2>
        Lorem ipsum
      </h2>
      <h4>
        Tomek S
      </h4>
      <p>
         Lorem ipsum lorem ipsum
      </p>
    </div>
  </div>
</div>
`;
